src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpPreferFrameworkTypeDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpPreferFrameworkTypeDiagnosticAnalyzer : 
+        PreferFrameworkTypeDiagnosticAnalyzerBase<SyntaxKind, ExpressionSyntax, PredefinedTypeSyntax>
+    {
+        protected override ImmutableArray<SyntaxKind> SyntaxKindsOfInterest => 
+            ImmutableArray.Create(SyntaxKind.PredefinedType);
+
+        protected override bool IsPredefinedTypeReplaceableWithFrameworkType(PredefinedTypeSyntax node)
+        {
+            var syntaxKind = node.Keyword.Kind();
+
+            // every predefined type keyword except `void` can be replaced by its framework type in code.
+            return syntaxKind != SyntaxKind.VoidKeyword && SyntaxFacts.IsPredefinedType(syntaxKind);
```

Comment:
Yup! And if you remove that, you can inline `syntaxKind`. And _then_, it can be an expression bodied member. #prettycode